cstar/base/scheduler_job.py

```diff
@@ -0,0 +1,287 @@
+## STIL TODO:
+#
+# Add PBSJob class
+# Write unit test module
+# Write docstrings and rtd pages
+
+import os
+import re
+import time
+import subprocess
+from abc import ABC, abstractmethod
+from pathlib import Path
+from datetime import datetime
+from typing import Optional
+from cstar.base.system import cstar_system
+
+from cstar.base.scheduler import SlurmScheduler
+
+
+def create_scheduler_job(
+    commands: str,
+    cpus: int,
+    account_key: str,
+    script_path: Optional[str | Path] = None,
+    run_path: Optional[str | Path] = None,
+    job_name: Optional[str] = None,
+    output_file: Optional[str | Path] = None,
+    queue_name: Optional[str] = None,
+    send_email: Optional[bool] = True,
+    walltime: Optional[str] = None,
+) -> "SchedulerJob":
+    if isinstance(cstar_system.scheduler, SlurmScheduler):
+        job_type = SlurmJob
+    else:
+        raise TypeError(
+            f"Unsupported scheduler type: {type(cstar_system.scheduler).__name__}"
+        )
+
+    return job_type(
+        scheduler=cstar_system.scheduler,
+        commands=commands,
+        cpus=cpus,
+        account_key=account_key,
+        script_path=script_path,
+        run_path=run_path,
+        job_name=job_name,
+        output_file=output_file,
+        queue_name=queue_name,
+        send_email=send_email,
+        walltime=walltime,
+    )
+
+
+class SchedulerJob(ABC):
+    @abstractmethod
+    def submit(self):
+        pass
```

Comment:
Would an enumeration of the possible statuses of a job be useful here?

```suggestion
class JobStatus(Enum):
    PENDING = auto()
    RUNNING = auto()
    COMPLETED = auto()
    FAILED = auto()


class SchedulerJob(ABC):
    @abstractmethod
    def submit(self):
        pass
    
    @property
    def status(self) -> JobStatus:
        return self._status
```

Reply:
Done in 907b767c24e687d7bf26448d6a668ffd674be5d4 and 23ab4d192f6299904fc71865b909f3311429ddd6